# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# Filename: package/.../ooo/fix_xmlhelp-port-to-db-4.3_diff.patch
# Copyright (C) 2006 The OpenSDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

--- ./patches/src680/xmlhelp-port-to-db-4.3.diff.orig	2006-10-19 18:20:32.000000000 -0400
+++ ./patches/src680/xmlhelp-port-to-db-4.3.diff	2006-10-19 18:21:47.000000000 -0400
@@ -118,9 +118,9 @@
  				// TODO Auto-generated catch block
  				e.printStackTrace();
  			}
---- xmlhelp/source/com/sun/star/help/HelpIndexer.java.ark	2005-07-23 01:22:18.000000000 +0200
-+++ xmlhelp/source/com/sun/star/help/HelpIndexer.java	2005-07-23 02:14:01.000000000 +0200
-@@ -151,12 +151,14 @@
+--- xmlhelp/source/com/sun/star/help/HelpIndexer.java~	2006-10-19 18:12:55.000000000 -0400
++++ xmlhelp/source/com/sun/star/help/HelpIndexer.java	2006-10-19 18:17:29.000000000 -0400
+@@ -124,12 +124,14 @@
  	private void schnitzel() {
  		// Determine the location of the database
  		String installDirectory = HelpDatabases.getInstallDirectory();
@@ -139,7 +139,7 @@
  
  			// Create indexDirectory, if not existent
  			String indexDirectory =
-@@ -191,16 +193,16 @@
+@@ -164,16 +166,16 @@
  			String fileName =
  				installDirectory + _language + File.separator + _module + ".db";
  
@@ -161,7 +161,7 @@
  				try {
  					String keyStr = key.getString();
  					String dataStr = data.getFile();
-@@ -229,13 +231,13 @@
+@@ -202,13 +204,13 @@
  				} catch (Exception e) {
  				}
  				if (first) {
@@ -177,8 +177,8 @@
 +			table.close(false);
  
  			System.out.println("Indexing...");
- 			Enumeration enum = _hashDocInfo.elements();
-@@ -314,10 +316,10 @@
+ 			Enumeration enumer = _hashDocInfo.elements();
+@@ -290,10 +292,10 @@
  			} catch (Exception e) {
  				System.err.println(e.getMessage());
  			}
@@ -191,22 +191,18 @@
  			System.out.println("Error initializing database");
  			System.exit(1);
  		} catch (FileNotFoundException fnfe) {
-@@ -429,9 +431,12 @@
- 				list[j++] = (String) enum.nextElement();
+@@ -405,9 +407,8 @@
+ 				list[j++] = (String) enumer.nextElement();
  			}
  
 -			Db table;
 +			Database table;
  			try {
 -				table = new Db(null, 0);
-+				DatabaseConfig cfg=DatabaseConfig.DEFAULT;
-+				cfg.setType(DatabaseType.BTREE);
-+				cfg.setAllowCreate(true);
-+				cfg.setMode(0644);
  
  				String fileName =
  					HelpDatabases.getInstallDirectory()
-@@ -440,15 +445,15 @@
+@@ -416,15 +417,15 @@
  						+ _module
  						+ ".key";
  
@@ -225,19 +221,22 @@
  			} catch (Exception e) {
  				System.out.println("error writing keydata");
  			}
-@@ -464,10 +469,8 @@
- 			list[j++] = (String) enum.nextElement();
+@@ -440,9 +441,12 @@
+ 			list[j++] = (String) enumer.nextElement();
  		}
  
 -		Db table;
-+		Database table;
++			Database table;
  		try {
 -			table = new Db(null, 0);
--
++				DatabaseConfig cfg=DatabaseConfig.DEFAULT;
++				cfg.setType(DatabaseType.BTREE);
++				cfg.setAllowCreate(true);
++				cfg.setMode(0644);
+ 
  			String fileName =
  				HelpDatabases.getInstallDirectory()
- 					+ _language
-@@ -475,15 +478,19 @@
+@@ -451,15 +455,19 @@
  					+ _module
  					+ ".ht";
  
